### ISSUE TITLE
Error Reporting Response Code

### ISSUE THREAD

**Issue by EatonZ** [googleapis/google-cloud-dotnet#7029]
In the Cloud Console Error Reporting section there is a Response Code column. Below is a screenshot with the hover tooltip.
Is it possible to populate it when logging an error?

![2021-08-23_0-13-24](https://user-images.githubusercontent.com/3020941/130389322-c71d8a01-512b-453f-8036-1073a5ab04e5.png)

**Comment by amanda-tarafa** [comment]
I think it should be possible. Will take a look and report back here when I know more, later this week.

**Comment by EatonZ** [comment]
@amanda-tarafa I log most exceptions myself using `IExceptionLogger.Log`, and in most cases will know the final HTTP status at that point. However, you are right in thinking the response code could change - anything can happen after the logging finishes, including a change to the final HTTP status. With that in mind, I think you are correct in thinking this wouldn't be worth the effort.

You have to think though - what was the error reporting team's intent behind that field?

**Comment by amanda-tarafa** [comment]
> You have to think though - what was the error reporting team's intent behind that field?

Although I cannot say with certainty what the intentions of the team's are, I think this field may be accurately used when the error logging software is external to the application producing the errors, say, a logging agent running on the environment that the application is running in. Also, not all applications are complex applications with a pipeline that may intercept a response, in those cases, you may more accurately log the Response Code from within the application itself. And, also, the Error Reporting component of Google Cloud is not just for logging application errors, it's used for logging infrastructure errors as well, say Compute instances failing to start up, Kubernetes pods not responding, etc. It's possible that the field may be used in those cases.
Basically, I think there are many use cases in which you could log a Response Code for an error, but there's just not a lot of overlap between those and the use cases covered by the Diagnostics libraries.

**Comment by EatonZ** [comment]
@amanda-tarafa Makes sense. I'll close this since it doesn't seem like it makes sense to add it to this library, based on how it works. Thanks for taking a look!
